waiter/integration/waiter/token_request_test.clj

```diff
@@ -1397,3 +1397,49 @@
                     (is (= service-id-3 (kitchen-env-service-id)))))))))
         (finally
           (delete-token-and-assert waiter-url token))))))
+
+(deftest ^:parallel ^:integration-fast test-valid-fallback-service-resolution
+  (testing-using-waiter-url
+    (let [service-name (rand-name)
+          token (create-token-name waiter-url service-name)
+          request-headers {:x-waiter-token token}
+          fallback-period-secs 300
+          token-description-1 (-> (kitchen-request-headers :prefix "")
+                                  (assoc :fallback-period-secs fallback-period-secs
+                                         :idle-timeout-mins 1
+                                         :name (str service-name "-v1")
+                                         :permitted-user "*"
+                                         :run-as-user (retrieve-username)
+                                         :token token
+                                         :version "version-1"))
+          token-description-2 (-> token-description-1
+                                  (assoc :name (str service-name "-v2")
+                                         :version "version-2")
+                                  (dissoc :cpus :mem :version))
+          token-description-3 (-> token-description-1
+                                  (assoc :name (str service-name "-v3")
+                                         :version "version-3"))]
+      (-> token-description-1
+          (as-> token-description (post-token waiter-url token-description))
+          (assert-response-status 200))
+      (let [service-id-1 (retrieve-service-id waiter-url request-headers)]
+        (with-service-cleanup
+          service-id-1
+          (let [response-1 (make-request-with-debug-info request-headers #(make-request waiter-url "/hello" :headers %))]
+            (assert-response-status response-1 200)
+            (is (= service-id-1 (:service-id response-1))))
+          (-> token-description-2
```

Comment:
Threading a small number of calls while also utilizing `as->` feels a bit awkward.  Would it be simpler to use `let` or write it normally?  e.g.  `(assert-response-status (post-token waiter-url token-description-2) 200))`

Reply:
Done.